### PR TITLE
[Pipeline](exec) support async writer in pipelien query engine

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/sink/async_result_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+std::unique_ptr<Block> AsyncResultWriter::get_block_from_queue() {
+    std::lock_guard l(_m);
+    DCHECK(!_data_queue.empty());
+    auto block = std::move(_data_queue.front());
+    _data_queue.pop_front();
+    return block;
+}
+
+void AsyncResultWriter::start_writer() {
+    ExecEnv::GetInstance()->fragment_mgr()->get_thread_pool()->submit_func(
```
Comment (yiguolei) [inline be/src/vec/sink/async_result_writer.cpp:57]: Maybe this need a lot of threads

**Comment by HappenLee** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.19 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          98.6 seconds inserted 10000000 Rows, about 101K ops/s
 storage size: 17162432315 Bytes

**Review by BiteTheDDDDt — APPROVED** [review]

File: be/src/vec/sink/async_result_writer.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            [this]() { this->process_block(); });
+}
+
+void AsyncResultWriter::process_block() {
+    if (!_is_open) {
+        _writer_status = open();
```
Comment (Gabriel39) [inline be/src/vec/sink/async_result_writer.cpp:79]: `process_block` will not be executed twice so we do not need to do this check

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**